tests/text/test_pipeline_with_custom_head.py

```diff
@@ -1,23 +1,37 @@
 import os
+from pathlib import Path
 from tempfile import mkdtemp
 
-from biome.text import Pipeline, PipelineConfiguration
-from biome.text.configuration import (
-    FeaturesConfiguration,
-    VocabularyConfiguration,
-)
-from biome.text.data import DataSource
-from biome.text.modules.heads import TaskHeadConfiguration, TextClassification
-from tests.test_context import TEST_RESOURCES
+import pytest
 
+from biome.text import Pipeline, PipelineConfiguration, Dataset
+from biome.text.configuration import FeaturesConfiguration,VocabularyConfiguration
+from biome.text.modules.heads import TaskHeadConfiguration, TextClassification
 
 class MyCustomHead(TextClassification):
     """Just a head renaming the original TextClassification head"""
-
     pass
 
+@pytest.fixture
+def training_dataset() -> Dataset:
+    """Creates the training dataset and gives the structure"""
+    resources_path = Path(__file__).parent.parent.parent / "tests" / "resources" / "data"
+    training_ds = Dataset.from_csv(paths=str(resources_path / "dataset_source.csv"))
+
+    # Keeping just 'label' and text 'category'
+    training_ds = training_ds.map(
+        lambda x: {"label": x["job"], "text": x["education"] + " " + x["marital"]}, 
+        remove_columns=["education", "marital"]
+    )
+    training_ds.remove_columns_(list(set(training_ds.column_names) - set(["label", "text"])))
```

Comment:
I would either remove all the columns in the `map` method, or remove the `remove_columns` argument from it. No need to remove columns twice. Actually, there is no need to remove the columns in the first place, you could just leave them in the dataset.